torch/distributed/optim/zero_redundancy_optimizer.py

```diff
@@ -121,17 +121,38 @@ def main_hook(self):
         self.zero.step()
 
 
-class _DDPBucket(NamedTuple):
+class _DDPBucketAssignment():
     r"""
-    This contains the model parameters corresponding to a
-    :class:`DistributedDataParallel` gradient bucket.
+    This represents a :class:`DistributedDataParallel` bucket assignment,
+    meaning a (possibly non-strict) subset of the parameters corresponding to
+    a DDP bucket assigned to a rank to update.
 
         bucket_index (int): index of the bucket determined by the DDP gradient
             bucket all-reduce order.
-        params (List[torch.Tensor]): model parameters in the bucket.
+        parameters (List[torch.Tensor]): model parameters in the bucket
+            assigned to this rank.
+        offset (int): offset into the :class:`GradBucket` 's :meth:`parameters`
+            giving the index of the first element in the passed-in
+            ``parameters``; this equivalently indexes into the
+            :class:`GradBucket` 's :meth:`gradients`.
+        device (torch.device): device on which the parameters are stored.
+        tensor (torch.Tensor): flattened tensor giving the data of the
+            parameter subset assigned to the rank.
```

Comment:
These two are attributes?

Reply:
Sorry, is it bad to have them as attributes?

Reply:
No, this is not a problem at all. I am just a bit confused at the first glance, and thought these are all arguments. 

The docstring for the method below has an argument section and an attribute section, which looks more clear.